### PR TITLE
Added more robust error handling

### PR DESCRIPTION
**Pull request by Logout22**
I improved on the corner cases of the script. Please do not hesitate to get back to me for feedback. Thank you!

### CODE REVIEW

File: build-image.sh [hunk 1]
```diff
@@ -37,24 +91,13 @@ chroot $chroot_dir apt-get -y install ubuntu-minimal
...
-
-### unmount /proc
-umount $chroot_dir/proc
+unmount_system_folders
 
 ### create a tar archive from the chroot directory
 tar cfz ubuntu.tgz -C $chroot_dir .
 
 ### import this tar archive into a docker image:
-cat ubuntu.tgz | docker import - $docker_image
```
Comment (dashohoxha) [inline build-image.sh]: Why did you comment this line?

Comment (Logout22) [inline build-image.sh]: Sorry, my bad. I used that comment for testing. Just reverted it.

**Comment by dashohoxha** [comment]
I do not understand the logic of the $safeguard, can you explain why it is needed and how it works?

**Comment by dashohoxha** [comment]
In general, did you have any problems with the script and tried to fix it? Or there were no problems but you are trying to make it better?
Are you able to test it? Because I don't have any i386 system anymore and so I cannot test it.

**Comment by Logout22** [comment]
Safeguard is a hard exit out of a potential endless while loop: If the system does not release the mount points, the shell script may never exit. Hence, the local variable $safeguard is initialized to the number of times that the unmount function will try to release the mount points before it gives up, and then consequently counted down after each try until it reaches zero and the script exits with an error.

**Comment by Logout22** [comment]
I tried the script on xenial first and had a lot of problems, which triggered the various error cases that are covered by the trap logic now. All those error cases worked as expected. Switching back to trusty/i386 and disabling the /dev mount point, I was also able to reproduce the script's happy path reliably (see commit above). However, for later distributions, I think the script would have to be changed in a backwards-incompatible way, mostly due to changes in udev, but as i386 is (hopefully) only required for legacy software, I think those changes and the accompanied testing effort could be postponed until support for trusty ends (ETA 2019).

**Comment by dashohoxha** [comment]
Thanks for explaining safeguard, I understand it now.

Do you think that creating a branch for ubuntu-16.04 (or xenial) will be useful? We can apply these changes only to this branch and leave the existing script untouched (or move it to the branch trusty).

Are you interested in becoming a maintainer of this repo? Since I am not able to test it, somebody else should take the responsibility of maintaining it.

**Comment by dashohoxha** [comment]
I am merging it and then will create a branch xenial.

**Comment by Logout22** [comment]
Thanks for merging!

Your second suggestion sounds more practical; I designed and tested my changes for trusty so they should be merged into master.

I would not mind maintaining this repo; I suppose that is a rather low-profile job, is it not?

**Comment by dashohoxha** [comment]
> I would not mind maintaining this repo; I suppose that is a rather low-profile job, is it not?

Right. Only when somebody submits any bug fix or PR, you have to test and make sure that it works, before accepting. This happens very rarely.

**Comment by dashohoxha** [comment]
@Logout22 I added you as an admin, so I guess that now you have full rights on the project.

**Comment by Logout22** [comment]
Thanks for adding me!

**Pull request merged**